### ISSUE TITLE
Data capturing

### ISSUE THREAD

**Issue by Paddy8** [tni/teensy-samples#5]
Hey there,

I got a problem implementing my Data capturing.
void captureData() {
  ret = Dps310PressureSensor.measurePressureOnce(pressure_dps310, oversampling);;
  logEntry( { pressure_dps310 } );
 }
If I am adding my Data capturing like this, the code never starts logging Data. It stops in Line 327 in the original code.

Any suggestions what I am doing wrong?

Best regards!

**Comment by imberest** [comment]
Attach your code to a txt file. I can take a look.
on line 327 it checks if buffer is filled, and if it is then it writes it to a SD card. It seems that you are not filling buffer.
TNI is slow on answering, busy man :)

**Comment by Paddy8** [comment]
Thank you for your time @imberest .
There is no problem if I am trying to capture analog data.

[SdDatSDIO.txt](https://github.com/tni/teensy-samples/files/2850457/SdDatSDIO.txt)

**Comment by Paddy8** [comment]
[SdDatSDIO.txt](https://github.com/tni/teensy-samples/files/2850466/SdDatSDIO.txt)
Sorry, uploaded the wrong file.

**Comment by imberest** [comment]
From first look it seems that you are blocking writing with **do while** cycle.
I would suggest to check it with if statement.
Also you don't have time reference. I would suggest adding RTC to your circuit.

**Comment by Paddy8** [comment]
I don´t think this is the problem, because everything works fine, if I am using a sensor with analog output. But in this case I need a sensor with digital output. 
Is it possible, that calling the sensor function:
Dps310PressureSensor.measurePressureOnce(pressure_dps310, oversampling);
is blocking buffer writing somehow?

But all in all thank you very much so far.

**Comment by imberest** [comment]
```
// ISR, periodically triggered by PIT timer. It creates some dummy
// log entries. Put your data capturing here.
void captureData() {
  //while (Serial2.available() > 0){
  //while(LED2 == HIGH) {
  Dps310PressureSensor.measurePressureOnce(pressure_dps310, oversampling);
  //Serial.println(pressure_dps310);
  //pressure_dps310;
  //sensor.measure(sensor.FAST);
  //sensor.getPressurePa();
  /*int analog_pin = analogRead(A18);
  digitalWrite(30, 1);
  delayMicroseconds(1);
  digitalWrite(30, 0);*/
  logEntry( { pressure_dps310 } );
  //}
record_offset += sizeof(LogEntry); // I think you are missing this one
}
```

But if i was you i would go with something like this:
Get rid of all **do while** and check on `captureData` void. 

```
// ISR, periodically triggered by PIT timer. It creates some dummy
// log entries. Put your data capturing here.
void captureData() {
    if(LED2 == HIGH){
      Dps310PressureSensor.measurePressureOnce(pressure_dps310, oversampling);
      logEntry( { pressure_dps310 } );
    record_offset += sizeof(LogEntry); // I think you are missing this one
    }
}
```

**Comment by imberest** [comment]
If you want i could help you with debugging, but would like to go off this comment thing. TNI will get tons of notifications, and this is not the best place to chat :) so if you have any ideas how we can get in touch privately, i will help you as much as i can.

**Comment by Paddy8** [comment]
Okay, thank you. Now there is a new problem, it starts writting to the SD Card, but no bytes are written.
elapsed: 9s     bytes written: 0     buffer overflow count: 0
next write sector: 54693948    max buffers used during last 5s: 0
elapsed: 14s     bytes written: 0     buffer overflow count: 0
next write sector: 54693948    max buffers used during last 5s: 0

**Comment by Paddy8** [comment]
Okay try p.jedlicka@gmx.de
